bin/wmagent-resource-control

```diff
@@ -95,32 +95,19 @@ def createOptionParser():
 
     return myOptParser
 
-def getCMSSiteInfo(pattern):
+def getCMSSiteInfo(pattern=''):
     """
     _getCMSSiteInfo_
 
     Query SiteDB for the site and SE names matching the pattern.  Return a
     dictionary keyed by site name.
     """
-    nameSEMapping = {}
     print "Querying SiteDB for site information..."
 
-    siteDB = SiteDBJSON()
-    for seName in siteDB.cmsNametoList(pattern, "SE", "result.json"):
-        cmsNames = siteDB.seToCMSName(seName)
-        for cmsName in cmsNames:
-            if cmsName in nameSEMapping:
-                nameSEMapping[cmsName].append(seName)
-            else:
-                nameSEMapping[cmsName] = [seName]
-
+    pattern = pattern.replace('*', '.*').replace('%', '.*')
     if options.addDisk:
-        nameSEMapping = dict((k, list(set(v))) for k, v in nameSEMapping.iteritems()
-                        if not k.endswith("_Disk"))
-    else:
-        nameSEMapping = dict((k, list(set(v))) for k, v in nameSEMapping.iteritems())
-
-    return nameSEMapping
+        return SiteDBJSON().PSNtoPNNMap('^(?!.*_Disk\s*$)' + pattern)
```

Comment:
The logic hasn't changed from before, it's just set up in a strange way such that if on --addDisk is given then it defaults to True otherwise it is False. Hence the backwards looking logic but it does work. I didn't change the logic as that wasn't crucial to the patch that I'm  implementing also I assumed that there must have been a reason it was done in this backwards way.